core/index/index_writer.hpp

```diff
@@ -202,11 +202,12 @@ class index_writer : private util::noncopyable {
     /// @brief create a document to filled by the caller
     ///        for insertion into the index index
     ///        applied upon return value deallocation
+    /// @arg no_flush don't trigger segment flush
     /// @note the changes are not visible until commit()
     ////////////////////////////////////////////////////////////////////////////
-    document insert() {
+    document insert(bool no_flush = false) {
```

Comment:
Maybe `flush` or `need_flush` instead of `no_flush`

Reply:
done